tests/test_flow.py

```diff
@@ -490,16 +493,19 @@ def test_dataframe_source_with_metadata():
     expected = [
         Event(
             {"my_key": "key1", "my_time": t1, "my_id": "id1", "my_value": 1.1},
-            key="key1",
+            key=key1,
             id="id1",
         ),
         Event(
             {"my_key": "key2", "my_time": t2, "my_id": "id2", "my_value": 2.2},
-            key="key2",
+            key=key2,
             id="id2",
         ),
     ]
-    assert termination_result == expected
+    assert len(termination_result) == len(expected)
+    for result, expected_event in zip(termination_result, expected):
+        assert result == expected_event
+        assert result.key == expected_event.key  # __eq__ method does not check key.
```

Comment:
Maybe better to check like this, so that we get a more useful error message (not e.g. `AssertionError: 1 != 2`)
```suggestion
    assert termination_result == expected
    assert list(map(lambda event: event.key, termination_result)) == [key1, key2]
```

Reply:
great idea!